ios/brave-ios/Sources/PlaylistUI/MediaContentView.swift

```diff
@@ -0,0 +1,287 @@
+  @Environment(\.isFullScreen) private var isFullScreen
+  @Environment(\.toggleFullScreen) private var toggleFullScreen
+  @Environment(\.requestGeometryUpdate) private var requestGeometryUpdate
+
+  var body: some View {
+    VStack {
+      PlayerView(playerModel: model)
+        .zIndex(1)
+        .playlistSheetDetentAnchor(id: .mediaPlayer)
+      if !isFullScreen {
+        PlaybackControlsView(model: model, selectedItemTitle: selectedItem.name)
+          .padding(24)
+      }
+    }
+    .frame(maxWidth: .infinity, maxHeight: .infinity, alignment: isFullScreen ? .center : .top)
+    .background(isFullScreen ? .black : Color(braveSystemName: .containerBackground))
+    .onChange(of: isFullScreen) { newValue in
+      // Automatically rotate the device orientation on iPhones when the video is not portrait
+      if UIDevice.current.userInterfaceIdiom == .phone, !model.isPortraitVideo {
+        requestGeometryUpdate(orientation: newValue ? .landscapeLeft : .portrait)
+      }
+    }
+    .onChange(of: interfaceOrientation) { newValue in
+      if UIDevice.current.userInterfaceIdiom == .phone {
+        if newValue.isLandscape {
+          // Always toggle fullscreen on landscape iPhone
+          toggleFullScreen(explicitFullScreenMode: true)
+        } else {
+          toggleFullScreen(explicitFullScreenMode: model.isPortraitVideo)
+        }
+      }
+    }
+    .persistentSystemOverlays(isFullScreen ? .hidden : .automatic)
+    .defersSystemGestures(on: isFullScreen ? .all : [])
+    .ignoresSafeArea(.keyboard, edges: .bottom)
+  }
+}
+
+@available(iOS 16.0, *)
+extension MediaContentView {
+  /// A media scrubber that controls the current player
+  ///
+  /// This is a separate view since it is constantly updating the current time, so its better to
+  /// keep that state isolated and only update this View when time changes instead of the entire
+  /// PlaybackControlsView
+  private struct PlaybackScrubber: View {
+    @ObservedObject var model: PlayerModel
+
+    @State private var currentTime: TimeInterval = 0
+    @State private var isScrubbing: Bool = false
+    @State private var resumePlayingAfterScrub: Bool = false
+
+    var body: some View {
+      MediaScrubber(
+        currentTime: Binding(
+          get: {
+            if isScrubbing {
+              return model.currentTime
+            } else {
+              return self.currentTime
+            }
+          },
+          set: { newValue in
+            Task { await model.seek(to: newValue, accurately: true) }
+          }
+        ),
+        duration: model.duration,
+        isScrubbing: $isScrubbing
+      )
+      .tint(Color(braveSystemName: .iconInteractive))
+      .onChange(of: isScrubbing) { newValue in
+        if newValue {
+          resumePlayingAfterScrub = model.isPlaying
+          model.pause()
+        } else {
+          if resumePlayingAfterScrub {
+            model.play()
+          }
+        }
+      }
+      .task(priority: .low) {
```

Comment:
Isnt it actually supposed to be high priority since it is changing  the current time?

Reply:
I think low is fine since it is a repeating task and only updates the UI label, the underlying player time will still always be correct